src/Platform/Microsoft.Testing.Platform/ServerMode/JsonRpc/ServerModeManager.cs

```diff
@@ -20,34 +18,10 @@ internal IMessageHandlerFactory Build(IServiceProvider serviceProvider)
             ? int.Parse(clientPortArgs[0], CultureInfo.InvariantCulture)
             : throw new InvalidOperationException(PlatformResources.MissingClientPortFoJsonRpc);
 
-        string? clientHostName;
-        clientHostName = commandLineService.TryGetOptionArgumentList(PlatformCommandLineProvider.ClientHostOptionKey, out string[]? clientHostArgs)
+        string clientHostName = commandLineService.TryGetOptionArgumentList(PlatformCommandLineProvider.ClientHostOptionKey, out string[]? clientHostArgs)
             ? clientHostArgs[0]
             : "localhost";
 
-        if (CommunicationProtocol is not null)
```

Comment:
Note to make it easier for review.

We never set `CommunicationProtocol`, and never instantiated `JsonRpcTcpServerToSingleClient`. So cleaned all that up.

The cleanup made it such that this method (`Build`), can be made static. So made it static, and consequently cleaned up the no longer needed interface `IServerModeManager` along with the `ServerMode` property that is no longer needed as the `Build` method is accessed directly from `ServerModeManager`